openstef/model/regressors/arima.py

```diff
@@ -0,0 +1,163 @@
+
+        Parameters
+        ----------
+        start : int, str, or datetime, optional
+            Zero-indexed observation number at which to start forecasting, i.e.,
+            the first forecast is start. Can also be a date string to parse or a datetime type.
+            Default is the the zeroth observation.
+        end : int, str, or datetime, optional
+            Zero-indexed observation number at which to end forecasting, i.e.,
+            the last forecast is end. Can also be a date string to parse or a datetime type.
+            However, if the dates index does not have a fixed frequency,
+            end must be an integer index if you want out of sample prediction.
+            Default is the last observation in the sample.
+        exog : pd.DataFrame
+            Exogenous data (features).
+        quantile : float
+            The quantile for the confidence interval.
+
+        Returns
+        -------
+        pd.Serie
+            The quantile prediction.
+
+        """
+        alpha = quantile
+        idx = 0
+        if quantile > 0.5:
+            alpha = 1 - quantile
+            idx = 1
+        return (
+            self.results_.get_prediction(start, end, exog=exog)
+            .conf_int(alpha=alpha)
+            .iloc[:, idx]
+        )
+
+    def predict(self, x, quantile: float = 0.5, **kwargs):
+        start = x.iloc[0].name
+        end = x.iloc[-1].name
+        exog = x
+        predictions = self.results_.predict(start, end, exog=exog).to_numpy()
+        if quantile != 0.5:
+            predictions = self.predict_quantile(start, end, exog, quantile)
+        return predictions
+
+    def set_feature_importance(self):
+        """Because report needs 'weight' and 'gain' as importance metrics, we set the values to these names.
+
+        - 'weight' is corresponding to the coefficients values
+        - 'gain' is corresponding to the pvalue for the nullity test of each coefficient
+
+        """
+        importances = pd.DataFrame(
+            {"weight": self.results_.params, "gain": self.results_.pvalues}
+        )
+        return importances
+
+    @property
+    def feature_names(self):
+        """The names of he features used to train the model."""
+        return self.feature_in_names_
+
+    @property
+    def can_predict_quantiles(self):
+        """Indicates wether this model can make quantile predictions."""
+        return True
+
+    def score(self, x, y):
+        """Compute R2 score for backtest prediction.
+
+        It need to update the historic with (X_past, y_past).
+
+        """
+        ys_true = []
+        ys_pred = []
+
+        freq = pd.infer_freq(x.index)
+        if not (freq[0].isdigit()):
+            freq = f"1{freq}"
+        test_size = pd.Timedelta(self.backtest_max_horizon, "minutes") // pd.Timedelta(
+            freq
+        )
+        n_splits = (x.shape[0] // test_size) - 1
+        tscv = TimeSeriesSplit(n_splits=n_splits, test_size=test_size)
```

Comment:
Please use the full term here and not the abbreviation. I think you mean time_series_cross_validation ?